### ISSUE TITLE
Design - Overlaying two datasets

### ISSUE THREAD

**Issue by aboydnw** [NASA-IMPACT/veda-ui#359]
## Description

We have heard from user research that the ability to overlay two datasets or overlay a dataset over some contextual information, would be really useful.
An example of this might be overlaying a raster dataset showing smoke levels with a vector dataset showing the point locations or boundaries of fires. 

There are other, less mutable information that would be useful, in addition to the example above about fires. For example, countries, states, and watersheds are boundaries that will likely not change very often (if at all). So there will be some information similar to that where scientists may want to overlay datasets on top of.

In addition, they may want to use this contextual information (and/or vector datasets?) to easily define boundaries for analysis (or comparison/exploration?).

**Comment by danielfdsilva** [comment]
Co-authored with @ricardoduplos 

From a user experience point of view this is a tricky issue because it requires presenting similar types of options in different ways.

Even though technical differences do exist, for a user there's little difference between a NO2 Avg raster dataset, fire starting points, or coastlines. They're all data shown on a map.

Our layer panel currently allows the user to select one layer to enable (and one only). The new dashboard direction aims to change this, but it is not yet the case.

Keeping the panel as is (allowing the user to choose only 1 dataset layer) and adding a new section where multiple different contextual layers can be enable (coastlines and fires for example), may lead to confusion. In the end they are all just layers, have very similar appearances, but behave differently. The datasets are affected by the date selector on the top, but the contextual layers are not. This may lead to the user wondering why it works this way, and whether the app is broken or they're doing something wrong. There is no real explanation about why this would have to work this way.

Notwithstanding, we can move forward with such a solution knowing that it has limitations, takes us only hal-way home, and that the UX is subpar. This would allows us to have something done quickly enough, while the new direction is being finalized/implemented.


![Image](https://user-images.githubusercontent.com/1090606/212061102-18dc463f-a0da-4e50-9a18-788daaaa0b18.png)


Next step is to have a technical discussion about how the contextual layers would be configured by the users. cc @hanbyul-here @nerik

**Comment by aboydnw** [comment]
@ricardoduplos @danielfdsilva now that you have gone through this exercise, do you still feel we could learn something by working on this type of feature knowing that a redesign of the exploration/analysis workflow is coming? Said another way, would we benefit from continuing this work or will it just be throw-away? (feel free to hold off on this question until after a technical discussion if needed)

**Comment by danielfdsilva** [comment]
@aboydnw There is some work behind this feature and depending on the extent of the redesign/refactor there may be a lot that is discarded. However I don't think it is so complicated that development should be stopped. The questions I'd pose are:
Is this a feature that will be valuable for the users to have while we implement the redesign? Do we already have the non mapbox data that will be used as overlay?

**Comment by aboydnw** [comment]
Here is my opinion that is informed a bit by feedback sessions this week. I am really curious to hear if @ricardoduplos and @faustoperez would agree.

> Is this a feature that will be valuable for the users to have while we implement the redesign? 

My sense is this feature is MOST valuable to scientists in the context of the analysis workflow. That is when it usually comes up in conversation with them. It sounds like they would use these contextual layers to help them define an AOI. So, since this is just exploration at the moment, it might not be as valuable to them right now (maybe we can check this using the screenshot above?) However, building a feature like this in dataset exploration could afford us an opportunity to learn how users might use a feature like this. So, it may be more valuable to us as a learning tool than to users as a feature.

> Do we already have the non mapbox data that will be used as overlay?

We do not. Although, I don't think it would be difficult for us obtain. One that comes to mind is watersheds that are published by USDA, here: https://data.nal.usda.gov/dataset/watershed-boundary-dataset-wbd 
This could at least enable a proof-of-concept for a feature like this. I would imagine other contextual layers (like active fires) would essentially be data products within VEDA that are somehow marked as able to be used as contextual layers.

tl;dr I don't think the value we would gain from learning about how users would use this feature would outweigh the value of other development work (like a global search of datasets) if most of this work will be throwaway. I wonder if we could learn those things just from the screenshot you have in this ticket. I also don't think the design thinking that went into this ticket is throwaway either. We should show Brian and stakeholders that we did some thinking on this problem and let them know what our conclusions are. 

But I really want to know if you all see things differently after working on this ticket this sprint.

**Comment by hanbyul-here** [comment]
I think the layer toggling design and mechanism will be reused in some way. But it might help us to narrow down the feature only to Mapbox labels (and other currently available datasets) for now not to complicate it. (Focusing more on offering geospatial context through layering) ex. vector data support is another whole task that we haven't tackled yet.

**Comment by faustoperez** [comment]
> Is this a feature that will be valuable for the users to have while we implement the redesign?
I don't think it will be a throwaway, any map functionality we add now is compatible with the redesign.

@danielfdsilva Though technically they're all data layers, the wireframes proposal separates the context layers from the datasets, tying the Mapbox styles, labels, etc. together with other map tools such as uploading or drawing shapes. The two logical groups of the new analysis page are:

Group 1 (MAP VISUALIZATION) for:
- Labels
- Map projection
- Basemap styles
- Watersheds, admin areas, etc.
- Coastlines
- Etc.

Group 2 (DATA) for:
- data layers from the VEDA catalog

![Multiple Datasets - Analysis 5](https://user-images.githubusercontent.com/12268629/212647737-ef403e71-5d71-4456-975a-8bb760f59d0c.png)

(hope

**Comment by nerik** [comment]
@faustoperez I think you hit send before finishing writing. 

![](https://media2.giphy.com/media/14tzynlw3wBjrO/giphy.gif?cid=ecf05e47yaivy585vy1ki8kraz8l2clkuyilsf1anoa1cgwd&rid=giphy.gif&ct=g)

From what I gather from @faustoperez and @hanbyul-here we limit the meaning of "contextual layers" to generic features not specifically related to the studied dataset, available everywhere independently of the current Area, and just giving... context - that would exclude things like fires IMHO. 
In which case, it's great in itself, and reusable, but not covering the first part of what @aboydnw is asking.

> Keeping the panel as is (allowing the user to choose only 1 dataset layer) and adding a new section where multiple different contextual layers can be enable (coastlines and fires for example), may lead to confusion. In the end they are all just layers, have very similar appearances, but behave differently.

@ricardoduplos @danielfdsilva Could this be improved by moving context layers close to other generic map tools (projections/zoom/search etc) ?

![Screenshot 2023-01-16 at 11 21 55](https://user-images.githubusercontent.com/1583415/212655405-7ea680bb-040c-4c59-a06b-ccc014f458b7.png)

**Comment by danielfdsilva** [comment]
I think things get tricky when we try to define what is contextual layer.
Admin borders, map styles, labels, etc are definitely contextual information, easily related and associated with the map. These could be closer to other map tools and the interface would still be clear.

Things get murky when we talk about "contextual datasets" like the fires. These are not datasets per se (at least in the current dashboard version - which is what we're considering building this for) since they don't have a dedicated page and are meant to be viewed on top of other things. They may also have options, like the date selection, which is different from the global date selector. These are the problematic datasets which may cause confusion because they're very similar to the other ones.

We can define and even limit what is "contextual information". but it seems to me that, currently, we have 3 categories to worry about (better naming required):
- Real datasets (NO2, Agriculture, CO2, etc)
- Contextual datasets (Fires)
- Reference information (Borders, labels, Watersheds)

@nerik @faustoperez The new architecture will address this from the start, but for the time being how can we differentiate these 3 related yet different things?

**Comment by nerik** [comment]
Okay @danielfdsilva this is very useful, thanks. 

I suggest we treat __Reference information__ layers as an easy low-hanging fruit task and move it to another ticket @aboydnw 

__Contextual datasets__ on the other hand probably need to be better defined. What do they have in common? Do we have more examples beyond Fires? Could they be treated as _ancillary_ datasets ie attached to to a __real dataset__ ?

**Comment by danielfdsilva** [comment]
I would be great to get another example besides Fires.

I'm not sure we can attach it to a single real dataset. It may be useful to have it be contextual information for multiple. And if that's the case, for the time being, might as well be available everywhere as adding this restriction complicated development.

In any case, the way you control it (enable/disable/options) would be the same.

**Comment by faustoperez** [comment]
@danielfdsilva sounds good! Thanks for clarifying the current 3 categories.

I still see Fires belonging to the "real dataset" category, but for now, I think we'd be safe putting it as a context layer together with NO2 until we have a dedicated page for it. I agree with @nerik that we need to define the "contextual datasets" category, but that's something we can address in the exploration/analysis redesign.

What I would do, though, is to move the other "reference information" options closer to the map (maybe a separate ticket as @nerik suggests?), as they are global map visualization parameters independent of the active dataset.

**Comment by danielfdsilva** [comment]
> I think we'd be safe putting it as a context layer together with NO2 until we have a dedicated page for it

I don't think it should be tied to a single dataset. Fires for example may impact different measurements, like No2, Co2 and other air quality datasets. May also be relevant for Agriculture for example.

**Comment by faustoperez** [comment]
> I don't think it should be tied to a single dataset. Fires for example may impact different measurements, like No2, Co2 and other air quality datasets. May also be relevant for Agriculture for example.

Got it! Thanks! 😉 

Note to self: we'll need a way to feature context/related datasets or dataset groups when adding and viewing layers on the standalone "Exploration/Analysis" interface. I will try to consolidate these ideas here 👉 https://github.com/orgs/NASA-IMPACT/projects/17/views/6?pane=issue&itemId=18048591

**Comment by danielfdsilva** [comment]
Overlaying multiple datasets is a core feature of the Analysis and Exploration merge.
Relevant issue: https://github.com/NASA-IMPACT/veda-ui/issues/521

Closing.
